src/components/messages/create-group.js

```diff
@@ -0,0 +1,84 @@
+import React, {useEffect, useState} from "react";
+import * as service from "../../services/auth-service";
+import * as userService from "../../services/users-service";
+import {useNavigate} from "react-router-dom";
+import "./index.css";
+import MultiSelectUsers from "../multi-select-users";
+import * as groupService from "../../services/groups-service";
+
+
+/**
+ * Component to represent Group creation screen
+ * @returns {JSX.Element} React component
+ * @constructor
+ */
+const CreateGroup = () => {
+    const navigate = useNavigate();
+    const [allUsers, setAllUsers] = useState([]);
+    const [selectedUsers, setSelectedUsers] = useState([]);
+    const [group, setGroup] = useState(null);
+
+
+    useEffect(() => {
+        service.profile().then(me => {
+            userService.findAllUsers().then(userList => {
+                setAllUsers(userList.filter(user => user._id !== me._id));
+            });
+        }).catch(e => {
+            navigate('/login', {
+                state: {
+                    redirect: '/messages',
+                }
+            });
+        })
+    }, []);
+
+
+    /**
+     * Handler for user selection change
+     * @param users array of selected users
+     */
+    const handleChange = (allUsers) => {
+        setSelectedUsers(allUsers);
+        console.log(selectedUsers)
+    }
+
+    const handleGroupName = (e) => {
+        setGroup({...group, name: e.target.value});
+    }
+
+    const createGroup = () => {
+        if(selectedUsers.length < 1){
+            alert("Please select one or more users to add to your group.");
+            return;
+        }
+        groupService.createGroup("me", selectedUsers, group)
```

Comment:
We should ideally be sending a list of user ids and not list of users because the state of a user object can change at any point in the backend, but it's id will remain the same. Maybe something like this `selectedUsers.map(selectedUser => selectedUser._id)`

Reply:
I originally tried this, but it did not work.

Reply:
Can you take a look at line 52 of create-group component? This is what I originally tried to do, but it didn't work. Why can't I make group.users the list of ids?